src/main/java/com/petros/bring/bean/factory/AnnotationBeanFactory.java

```diff
@@ -27,6 +27,10 @@ public AnnotationBeanFactory(BeanDefinitionRegistry registry, List<BeanPostProce
 
     @Override
     public <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException {
+        Set<BeanDefinition> beanDefinitions = registry.getBeanDefinitionByType(beanType);
+        if (beanDefinitions.size() > 1 ) {
+            ensureBeanDefinitionsCreated(beanDefinitions);
```

Comment:
Looks like, here we create beans and method createBean(Class<?> clazz) is never used.

Reply:
When context is initialised, all our beans are instantiated. So this code can be useful only during application startup. Am I right?

Reply:
this snipped I added for the case when many (>1) beans are found by incoming type (interface) to address case when @Primary bean metadata exists but in context map is yet created.